### ISSUE TITLE
Lint/ToEnumArguments: False positive on `to_enum(__callee__)`

### ISSUE THREAD

**Issue by pdobb** [rubocop-hq/rubocop#9049]
Given the following code, Lint/ToEnumArguments reports an offense:

```ruby
def my_method
  return to_enum(__callee__) unless block_given?
  # ...
end

# Offense:
Lint/ToEnumArguments: Ensure you correctly provided all the arguments
  return to_enum(__callee__) unless block_given?
         ^^^^^^^^^^^^^^^^^^^
```

Whereas, in the same method, just changing `__callee__` to the name of the method passes:

```ruby
def my_method
  return to_enum(:my_method) unless block_given?
  # ...
end

# Passes ^
```

--------

## Expected behavior

Since [`__callee__`](https://ruby-doc.org/core-2.6.5/Kernel.html#method-i-__callee__) 
> Returns the called name of the current method as a Symbol. If called outside of a method, it returns nil.

... this should be an acceptable (and even preferred) method of using `to_enum`, because it allows the method name to be changed without requiring a change to the internals of the method to match.

## Actual behavior

A somewhat misleading offense is detected/reported:
> Ensure you correctly provided all the arguments

## Steps to reproduce the problem

Just add this method to a ruby file and run Lint/ToEnumArguments on it:

```ruby
def my_method
  to_enum(__callee__)
end

def my_method2
  to_enum(:my_method2)
end
```

Both should pass, but only the 2nd one does.

## RuboCop version

```
$ rubocop -V
1.3.0 (using Parser 2.7.2.0, rubocop-ast 1.1.1, running on ruby 2.6.6 x86_64-darwin19)
  - rubocop-performance 1.8.1
  - rubocop-rails 2.8.1
  - rubocop-rake 0.5.1
```

**Comment by marcandre** [comment]
Agreed.
I've been using `__method__` but you are right, `__callee__` can only be better. In contrived situations, `__method__` does not work where `__callee__` does:

```ruby
class Foo
  def foo
    return to_enum __method__ unless block_given?
    yield 42
  end

  alias bar foo
  undef foo
end

Foo.new.bar.to_a # => undefined method `foo' for #<Foo:0x00007f9a81a723a8>)
```
